fluent-plugin-enhance-k8s-metadata/lib/fluent/plugin/filter_enhance_k8s_metadata.rb

```diff
@@ -6,7 +6,91 @@ module Plugin
     class EnhanceK8sMetadataFilter < Fluent::Plugin::Filter
       Fluent::Plugin.register_filter('enhance_k8s_metadata', self)
 
+      require_relative '../../sumologic/kubernetes/cache_strategy.rb'
+
+      helpers :record_accessor
+      include SumoLogic::Kubernetes::Connector
+      include SumoLogic::Kubernetes::Reader
+      include SumoLogic::Kubernetes::CacheStrategy
+
+      # parameters for read/write record
+      config_param :in_namespace_path, :string, default: '$.namespace'
+      config_param :in_pod_path, :string, default: '$.pod'
+      config_param :out_root, :string, default: 'kubernetes'
+
+      # parameters for connecting to k8s api server
+      config_param :kubernetes_url, :string, default: nil
+      config_param :apiVersion, :string, default: 'v1'
+      config_param :client_cert, :string, default: nil
+      config_param :client_key, :string, default: nil
+      config_param :ca_file, :string, default: nil
+      config_param :secret_dir, :string, default: '/var/run/secrets/kubernetes.io/serviceaccount'
+      config_param :bearer_token_file, :string, default: nil
+      config_param :verify_ssl, :bool, default: true
+      # if `ca_file` is for an intermediate CA, or otherwise we do not have the
+      # root CA and want to trust the intermediate CA certs we do have, set this
+      # to `true` - this corresponds to the openssl s_client -partial_chain flag
+      # and X509_V_FLAG_PARTIAL_CHAIN
+      config_param :ssl_partial_chain, :bool, default: false
+
+      config_param :cache_size, :integer, default: 1000
+      config_param :cache_ttl, :integer, default: 60 * 60
+
+      def configure(conf)
+        super
+        normalize_param
+        connect_kubernetes
+        init_cache
+        @in_namespace_ac = record_accessor_create(@in_namespace_path)
+        @in_pod_ac = record_accessor_create(@in_pod_path)
+      end
+
       def filter(tag, time, record)
+        decorate_record(record)
+        record
+      end
+
+      private
+
+      def decorate_record(record)
+        namespace_name = @in_namespace_ac.call(record)
+        pod_name = @in_pod_ac.call(record)
+        if namespace_name.nil?
+          log.debug "Record doesn't have [#{@in_namespace_path}] field"
+        elsif pod_name.nil?
+          log.debug "Record doesn't have [#{@in_pod_path}] field"
+        else
```

Comment:
So we will only get label for pod, but not for other objects?

Reply:
yes